packages/glimmer-runtime/lib/compiled/opcodes/vm.ts

```diff
@@ -306,6 +313,77 @@ export class EvaluateOpcode extends Opcode {
   }
 }
 
+export class EvaluatePartialOpcode extends Opcode {
+  public type = "evaluate-partial";
+  public compiler: CompileInto & SymbolLookup;
+  public name: CompiledExpression<any>;
+
+  constructor({ name, compiler }: { compiler: CompileInto & SymbolLookup, name: CompiledExpression<any> }) {
+    super();
+    this.name = name;
+    this.compiler = compiler;
+  }
+
+  evaluate(vm: VM) {
+    let reference: PathReference<any> = this.name.evaluate(vm);
+    let name: InternedString = reference.value();
+
+    let { serializedTemplate } = vm.env.lookupPartial([name]);
+    let scanner = new Scanner(serializedTemplate, vm.env);
+    let block = scanner.scanInlineBlock(this.compiler.getSymbolTable());
+    vm.invokeBlock(block, vm.frame.getArgs());
+
+    if (!isConst(reference)) {
+      vm.pushDynamicScope();
+
+      vm.updateWith(new EvaluateUpdatingPartialOpcode({
+        compiler: this.compiler,
+        nameReference: reference,
+        name
+      }));
+    }
+  }
+
+  toJSON(): OpcodeJSON {
+    return {
+      guid: this._guid,
+      type: this.type,
+      args: []
+    };
+  }
+}
+
+
+export class EvaluateUpdatingPartialOpcode extends UpdatingOpcode {
+  public type = "evaluate-updating-partial";
+  public compiler: CompileInto;
+  public name: InternedString;
+  public nameReference: PathReference<any>;
+
+  constructor({ name, compiler, nameReference }: { name: InternedString, compiler: CompileInto & SymbolLookup, nameReference: PathReference<any> }) {
+    super();
+    this.compiler = compiler;
+    this.nameReference = nameReference;
+    this.name = name;
+  }
+
+  evaluate(vm: UpdatingVM) {
+    let name = this.nameReference.value();
+
+    if (name !== this.name) {
+      vm.throw();
```

Comment:
I feel like this is not the right way to force a re-render when the partial name changes